vulkaninfo/vulkaninfo.cpp

```diff
@@ -159,7 +159,7 @@ void DumpSurfaceCapabilities(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurf
     chain_iterator_surface_capabilities2(p, inst, gpu, surface.surface_capabilities2_khr.pNext);
 }
 
-void DumpSurface(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurface &surface, std::vector<std::string> surface_types) {
+void DumpSurface(Printer &p, AppInstance &inst, AppGpu &gpu, AppSurface &surface, std::unordered_set<std::string> surface_types) {
```

Comment:
I'm a little concerned about how the unordered set works with the loop on surface types in line 174.  If the set is truly unordered and no consistent iteration order is guaranteed, this could mean that the output could be different from run to run where there is no actual change.

I'd like to be able to assume that consecutive runs on exactly the same data should be guaranteed to produce exactly the same results.

Reply:
Right, unordered_set is my goto for performance reasons, but here the ordering of set is more important.